### ISSUE TITLE
Ability to create a .gif animation

### ISSUE THREAD

**Issue by Netroxen** [lupoDharkael/flameshot#172]
This is very useful in my web development, especially when creating tickets for issues on the frontend, however sometimes I'd like to demonstrate a usability or animation issue, for this it would be epic if there was a .gif creator. I little record button after making my selection with a configurable framerate in the settings would do wonders!

**Comment by lidel** [comment]
@Netroxen for now try [Peek](https://github.com/phw/peek) (a standalone animated GIF recorder).
But I agree, would be super neat to have it all in one tool.

**Comment by Netroxen** [comment]
@lidel yup I've actually been using Peek for a while, but this tool is absolutely the best I've found for quickly annotating and taking screenshots, so a GIF recorder would elevate this to *mind blown*.

I'll keep using Peek for the moment though...

**Comment by Bios-Marcel** [comment]
tip for those using a tiling window manager: Stay away from peek, it is unusable, instead use this:

```shell
#!/usr/bin/env sh

[ $# -lt 2 ] && \
  echo 'You have to pass a duration in seconds and a filename: "gif.sh 10 /tmp/record.gif"' && \
  exit 1

byzanz-record                                        \
  --cursor                                           \
  --verbose                                          \
  --delay=2                                          \
  --duration=${1}                                    \
  $(xrectsel "--x=%x --y=%y --width=%w --height=%h") \
  "${2}"
```

**Comment by bhayward93** [comment]
Much appreciated @Bios-Marcel , you've saved me quite the headache.

**Comment by NatoBoram** [comment]
@lidel Peek [doesn't work](https://github.com/phw/peek/issues/453) at the moment for recording GIFs.

**Comment by MichaelTunnell** [comment]
I dont think flameshot intends to be a video recorder so it would be better to check out something else such as SimpleScreenRecorder.

SSR does support saving as a GIF but honestly, DONT do that. You should always record as a video, preferably MKV or MP4, and then convert that to a GIF once you have it perfect for the GIF. Recording straight to GIF will lose so much quality and be gigantic filesizes.

In fact, I would suggest abandoning GIFs all together for big captures as direct video play is now available on most browsers even most mobile browsers via HTML5 so you dont need the GIF format anymore.

Even sites like Imgur default to Videos instead of GIFs because of this reason.

**Comment by Fedelaus** [comment]
To follow up on @Bios-Marcel

I'm using i3 and Peek, it works fine! Not sure if it's a change with Peek or if Bios was using a different tiling wm.

It's a floating window so you'll obviously need to consider that when using it, and if you were smart about it you could probably nicely control the window size/position via i3 commands to emulate full-screen/full-window etc.

**Comment by nitrocode** [comment]
Here are some gif tools on OSX

https://github.com/onmyway133/GifCapture
https://github.com/NickeManarin/ScreenToGif

```
brew install gifcapture
```

**Comment by adamscafuto** [comment]
I would throw in my two cents and point out another screenshot program (Gyazo) has a ~10 second GIF capture feature, so why would Flameshot not have it?

**Comment by NatoBoram** [comment]
Oh hey old thread but if you record a video, you can convert it to GIF *while keeping the quality*. It's just that `ffmpeg` doesn't do it by default.

```sh
ffmpeg -i input.mp4 -vf "split[s0][s1];[s0]palettegen[p];[s1][p]paletteuse" -loop 0 output.gif
```

**Comment by gladykov** [comment]
Peek is being deprecated because (along other reasons) author thinks GIF is a thing of a past, since video is now widely supported: https://github.com/phw/peek/issues/1191

That said, I would love to see video support in Flameshot, because it has such a nice interface. My benchmark is Windows program called Screenpresso, which could also visually show mouse clicks - very helpful thins for QAs . Right now I'm using Kazam next to Flameshot. It is simple but misses customization and Flameshot overlay UI, so Kazam + Flameshot would be a perfect blend.

**Comment by Doonut** [comment]
I have used flameshot for longer than I remember. I have it bound to a keybind calling flameshot gui. I think with that, there are tons of mountains to cross because of how flameshot works to implement this, like completely freezing the window when I call that command which I very do use and like. For example, if I were to call that command, the window freezes, then I take time to resize my window, etc. what about the actual content that changed in the last x seconds and I want a gif? But then if you add that functionality, what about the SPEED of the application you love that disappears because it needs to buffer the last y seconds JUST IN CASE you take a video? I would absolutely ADORE this functionality, but I believe it should at LEAST be a separate call, if not a fork of flameshot with another name, so it could only focus on video and be a separate instead of what flameshot currently does, which is implement everything you want in a screenshot and be VERY fast about it.

**Comment by zhaynoth** [comment]
Having would be great, it is very useful on development/testing communications

**Comment by heyakyra** [comment]
[Kooha](https://github.com/SeaDve/Kooha) supports gif and you can even use a modern codec like [AV1](https://github.com/SeaDve/Kooha/issues/151)

**Comment by r0bag** [comment]
**+1**

**Comment by ragnarok22** [comment]
The Peek project has been [declared deprecated](https://github.com/phw/peek/issues/1191) 😢

**Comment by glaudiston** [comment]
This would be nice to have. Here the command I use to record part of screen:
```
ffmpeg -f x11grab -video_size 1180x928 -framerate 50 -i :0.0+927,130 -vf format=yuv420p output.mp4
```

**Comment by nealoke** [comment]
This would be so nice as the interface is already super fast and sleek 🥹, Windows' snip tool also has a toggle to capture image or video which saves a ton of time. 

PS: Just notices that SHIFT + WINDOWS + R starts the recording in the window' snip tool :)

**Comment by TheInfamousAlk** [comment]
Still hoping for this. Windows has ShareX which has this, and I had to move to flameshot when I moved to Linux. I'm definitely missing gif recording in the same interface as my screenshot tool.

**Comment by negativems** [comment]
Waiting for this feature too 👀

**Comment by PC-SPEZIALIST** [comment]
I suppose video recording is out of scope as already mentioned but you may have a look at [Screenpresso](https://www.screenpresso.com/), @godalming123.

**Comment by gladykov** [comment]
> I suppose video recording is out of scope as already mentioned but you may have a look at [Screenpresso](https://www.screenpresso.com/), [@godalming123](https://github.com/godalming123).

Screenpresso ia a great paid tool only for Windows, so recommending it in an open source cross-platform tool is not best choice of action.

**Comment by Wranglyph** [comment]
I second the idea of a companion tool- too many gismos clutters the experience. "Flowshot," maybe, idk.
Heck, you could even have a manager tool that lets you assign which gizmos belong to which "__shot"- which would also open up the ability for some kind of community library, chock full of niche tools that wouldn't make sense in the main flameshot ux.
